### DISCUSSION TITLE
Jumplist entries for using ex-mode for line navigation.

### DISCUSSION THREAD

**Discussion by askreet** [helix-editor/helix#5741] [Q&A] [answered]
When I use, for example, `:123` to navigate to line number 123 Helix will jump to line 1, then line 12 and finally line 123 as I type. Each time it navigates as I type an entry is placed into the jumplist so that once I'm done with line 123 pressing C-o jumps to line 12, then line 1, then finally back to where I was before I started my journey.

A couple questions for the community:
1. Is there a configuration option to disable jumping as a number is typed? I actually find it quite jarring in a large file to jump way to the top, then progressively back toward where I want to go.
2. Separately, is it possible _not_ to add the intermediate lines to the jumplist?
3. Is all of this the intended behavior, or something that should land in an Issue?

Thanks in advance!

**Comment by the-mikedavis** [comment, marked as answer]
The code for this function is here: https://github.com/helix-editor/helix/blob/c9b583ea9b59aa78c12e3831ed9c73f83d1d35d8/helix-term/src/commands/typed.rs#L1413-L1455

The `goto_line_impl` helper pushes an entry to the jumplist. For this case we should only push a new jumplist entry when the `PromptEvent` is `PromptEvent::Validate` (happens when you hit Return).

There isn't currently a way to disable this behavior. It would be nice to debounce it, maybe using the idle-timeout.

**Reply by askreet** [reply]
Thanks, Mike. I'm going to try my hand at adding this.

**Reply by askreet** [reply]
Cross-linking #5751 as resolving the intermediate bit.

**Comment by CptPotato** [comment]
I personally like to use `123gg` (alternatively `g123g`) to jump to a specific line.

It doesn't have a preview, though.

**Reply by askreet** [reply]
Thanks! I knew there was probably another key combo for this. 👍🏼
